pkg/ipam/node.go

```diff
@@ -645,6 +650,15 @@ func (n *Node) updateLastResync(syncTime time.Time) {
 // MaintainIPPool attempts to allocate or release all required IPs to fulfill
 // the needed gap. If required, interfaces are created.
 func (n *Node) MaintainIPPool(ctx context.Context) error {
+	// As long as the instances API is unstable, don't perform any
+	// operation that can mutate state.
+	if !n.manager.InstancesAPIIsReady() {
+		if n.retry != nil {
+			n.retry.Trigger()
+		}
+		return fmt.Errorf("instances API is unstable. Blocking mutating operations. See logs for details.")
+	}
+
```

Comment:
This chunk makes unit tests fail. As far as I understand this is because API has never been sync'ed in the first place, and synchronisation is called at the end of this function (`n.manager.resyncTrigger.Trigger()`).

Adding `n.manager.resyncTrigger.Trigger()` just above the chunk works but is probably not the expected fix... It should more likely be part of `n.retry.Trigger()`? Maybe something like this, for the last chunk of this set in `pkg/ipam/node_manager.go`:

``` go
		retry, err := trigger.NewTrigger(trigger.Parameters{
			Name:        fmt.Sprintf("ipam-pool-maintainer-%s-retry", resource.Name),
			MinInterval: time.Minute, // large minimal interval to not retry too often
			TriggerFunc: func(reasons []string) {
				poolMaintainer.Trigger()
				n.resyncTrigger.Trigger() // NOT SURE IF CORRECT, BUT FIXES UNIT TESTS
			},
		})
		if err != nil {
			node.logger().WithError(err).Error("Unable to create ...")
			return false
		}
		node.retry = retry
```